azure-keyvault/examples/test_examples_keyvault.py

```diff
@@ -0,0 +1,197 @@
+
+from azure.keyvault.secrets._client import SecretClient
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.core.exceptions import ClientRequestError
+
+def create_vault_client():
+    # [START create_vault_client]
+    import os
+    from azure.keyvault.vault_client import VaultClient
+
+    vault_url = 'https://myvault.vault.azure.net.'
+    credentials =  ServicePrincipalCredentials(
+        client_id = client_id,
+         secret = client_secret,
+        tenant = tenant_id,
+        resource = "https://vault.azure.net"
+    )
+
+    # Create a new Vault client using Azure credentials
+    vault_client = VaultClient(vault_url=vault_url, credentials=credentials)
+    # [END create_vault_client]
+    return vault_client
+
+
+def test_example_secret_crud_operations():
+   
+    vault_client = create_vault_client()
+    client = vault_client.secrets
+
+    try:
+        # [START get_secret]
+
+        # get secret with version
+        client.get_secret('secret name', 'secret_version')
+
+        # [END get_secret]
+    except ClientRequestError:
+        pass
+
+    try:
+        # [START set_secret]
+        from dateutil import parser as date_parse
+
+        expires = date_parse.parse('2050-02-02T08:00:00.000Z')
+        client.set_secret('secret name', 'secret_value', enabled=False, expires=expires)
+
+        # [END set_secret]
+    except ClientRequestError:
+        pass
+
+    try:
+        # [START update_secret_attributes]
+
+        content_type = 'text/plain'
+        expires = date_parse.parse('2050-02-02T08:00:00.000Z')
+        tags = {'foo': 'updated tag'}
+        
+        secret_bundle = client.update_secret_attributes(
+            'secret name', 'secret_version',
+            content_type=content_type,
+            expires=expires,
+            tags=tags)
+
+        # [END update_secret_attributes]
+    except ClientRequestError:
+        pass
+
+def test_example_secret_list_operations():
+
+    vault_client = create_vault_client()
+    client = vault_client.secrets
+
+    try:
+        # [START list_secrets]
+
+        # get secret with max_page_size
+        client.list_secrets(max_page_size=3)
+
+        # get secret without max_page_size (default = 25)
+        client.list_secrets()
+
+        # [END list_secrets]
+    except ClientRequestError:
+        pass
+
+    try:
+        # [START list_secret_versions]
+        import os
+        from azure.keyvault.vault_client import VaultClient
+        
+        # get secret with max_page_size
+        client.list_secret_versions('secret name', max_page_size=26)
+
+        # get secret without max_page_size
+        client.list_secret_versions('secret name')
+
+        # [END list_secret_versions]
+    except ClientRequestError:
+        pass
+
+    try:
+        # [START list_deleted_secrets]
+        import os
+        from azure.keyvault.vault_client import VaultClient
+        # TODO: should we put here how to create a vault client first?
```

Comment:
It's okay to assume a client already exists, creation will be shown in its own example.